internal/operators/lvm/lvm_operator.go

```diff
@@ -118,14 +122,31 @@ func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, ho
 		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, err
 	}
 
-	// GetValidDiskCount counts the total number of valid disks in each host and return an error if we don't have the disk of required size
 	diskCount := o.getValidDiskCount(inventory.Disks, host.InstallationDiskID)
-	if err != nil {
-		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{err.Error()}}, nil
+	mode := o.GetDeploymetMode(cluster)
+
+	message := "Logical Volume Manager requires at least one non-installation HDD/SSD disk on the host"
+	if mode == common.ClusterDeploymentModeCompact || mode == common.ClusterDeploymentModeSNO {
+		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
+			if diskCount == 0 {
+				return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, nil
+			}
+			return api.ValidationResult{Status: api.Success, ValidationId: o.GetHostValidationID()}, nil
+		}
 	}
-	if diskCount == 0 {
-		message := "Logical Volume Manager requires at least one non-installation HDD/SSD disk on the host"
-		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, nil
+
+	if mode == common.ClusterDeploymentModeStandard {
+		if host.Role == models.HostRoleAutoAssign {
+			status := "For LVM Standard Mode, host role must be assigned to master or worker."
+			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{status}}, nil
+		}
+	}
+
+	// check worker hosts or  unbind hosts to cluster
+	if host.Role != models.HostRoleMaster {
```

Comment:
Do we know for a fact that this is a requirement?  LVMS won't manage if one worker doesn't have disks?

Reply:
The storage must be consistent across clusters so the requirement should be at least one disk per valid node.
 Assisted is enforcing one disk on each valid node.
The operator can be deployed without disks on all valid nodes.

Reply:
OK, if that's in fact the requirement, we should check it